crates/eips/src/eip7840.rs

```diff
@@ -87,9 +105,23 @@ impl BlobParams {
         &self,
         excess_blob_gas: u64,
         blob_gas_used: u64,
+        base_fee_per_gas: u64,
     ) -> u64 {
-        (excess_blob_gas + blob_gas_used)
-            .saturating_sub(eip4844::DATA_GAS_PER_BLOB * self.target_blob_count)
+        let block_gas_excess_and_used_sum = excess_blob_gas + blob_gas_used;
+        let target_blob_gas = self.target_blob_gas_per_block();
+        if block_gas_excess_and_used_sum < target_blob_gas {
+            return 0;
+        }
+
+        if BLOB_BASE_COST as u128 * base_fee_per_gas as u128
+            > DATA_GAS_PER_BLOB as u128 * self.calc_blob_fee(excess_blob_gas)
+        {
+            block_gas_excess_and_used_sum
+                + (blob_gas_used * (self.max_blob_count - self.target_blob_count)
```

Comment:
Should `max_blob_count` and `target_blob_count` be from the parent block or the current block (In this function current block or next block)? We are currently using parent block max/target count and I can't deduce what is specified in EIP